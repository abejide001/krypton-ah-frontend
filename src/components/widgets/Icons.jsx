import React from 'react';
import PropTypes from 'prop-types';
import { Menu, Icon, Rating } from 'semantic-ui-react';
import Reactions from './ReactionIcons';

export default function Icons(props) {
  const { id, ratingIcon, numberofcomments, date, name, rating, disabled } = props;
  return (
    <Menu className="widgetContainer">
      <div className="icons">
        {ratingIcon ? (
          <React.Fragment>
            <Reactions date={date} selectedArticleId={id} />
          </React.Fragment>
        ) : (
          <React.Fragment>
            <Reactions date={date} />
            <Icon disabled link size="small" name="bookmark outline" />
            <span>{numberofcomments}</span>
            <div className="rating">
              <Rating disabled={disabled} maxRating={5} defaultRating={rating} size="small" />
              {name ? <span>{name}</span> : ''}
            </div>
          </React.Fragment>
        )}
      </div>
    </Menu>
  );
}
Icons.defaultProps = {
  id: 1,
  name: 'author-name',
  date: 'current-date',
  ratingIcon: 'rating-class',
  numberofcomments: 23,
  rating: 0,
  disabled: false
};
Icons.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  date: PropTypes.string,
  ratingIcon: PropTypes.string,
  numberofcomments: PropTypes.number,
  rating: PropTypes.number,
  disabled: PropTypes.bool
};
