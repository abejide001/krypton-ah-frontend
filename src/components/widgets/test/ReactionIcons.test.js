import React from 'react';
import { shallow } from 'enzyme';
import Icons from '../ReactionIcons';

describe('Test for reaction Icons', () => {
  it('should have initial state of false and change when clicked ', () => {
    const wrapper = shallow(<Icons />);
    const likeButton = wrapper.find('i').at(1);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.state().like).toEqual(false);

    likeButton.simulate('click');
    expect(wrapper.state().like).toEqual(true);

    likeButton.simulate('click');
    expect(wrapper.state().like).toEqual(false);
  });

  it('should have initial state of false and change when clicked ', () => {
    const wrapper = shallow(<Icons />);
    const dislikeButton = wrapper.find('i').at(2);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.state().dislike).toEqual(false);

    dislikeButton.simulate('click');
    expect(wrapper.state().dislike).toEqual(true);

    dislikeButton.simulate('click');
    expect(wrapper.state().dislike).toEqual(false);
  });
});
