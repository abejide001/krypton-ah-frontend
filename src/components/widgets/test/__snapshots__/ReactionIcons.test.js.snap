// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for reaction Icons should have initial state of false and change when clicked  1`] = `
<Fragment>
  <Icon
    as="i"
    disabled={true}
    fitted={true}
    link={true}
    name="time"
    size="small"
  >
    <span>
      Invalid Date
    </span>
  </Icon>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs up thumbs up outline"
      size="small"
    />
  </i>
  <span>
    2
  </span>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs down thumbs down outline"
      size="small"
    />
  </i>
  <span>
    1
  </span>
  <i
    className="articleComment"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="comments outline"
      size="small"
    />
  </i>
  <span
    className="commentspan"
  >
    0
  </span>
</Fragment>
`;

exports[`Test for reaction Icons should have initial state of false and change when clicked  2`] = `
<Fragment>
  <Icon
    as="i"
    disabled={true}
    fitted={true}
    link={true}
    name="time"
    size="small"
  >
    <span>
      Invalid Date
    </span>
  </Icon>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs up thumbs up outline"
      size="small"
    />
  </i>
  <span>
    2
  </span>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs down thumbs down outline"
      size="small"
    />
  </i>
  <span>
    1
  </span>
  <i
    className="articleComment"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="comments outline"
      size="small"
    />
  </i>
  <span
    className="commentspan"
  >
    0
  </span>
</Fragment>
`;
